src/SAWScript/Builtins.hs

```diff
@@ -636,6 +637,50 @@ extract_uninterp unints opaques tt =
      pure (tt', replList)
 
 
+congruence_for :: TypedTerm -> TopLevel TypedTerm
+congruence_for tt =
+  do sc <- getSharedContext
+     congTm <- io $ build_congruence sc (ttTerm tt)
+     io $ mkTypedTerm sc congTm
+
+build_congruence :: SharedContext -> Term -> IO Term
```

Comment:
If I understand correctly, this builds a proposition in the Curry-Howard sense, not the typical SAW boolean-valued function sense. This might be worth mentioning in the documentation. Is the use case here to build a lemma for the rewriter?

Reply:
That's correct, this is a Curry-Howard style proposition.

The use case here is mostly to directly apply the resulting lemma using `goal_apply`. The rewriter isn't really set up to make use of lemmas like this.